tenable/io/v3/vm/target_groups.py

```diff
@@ -0,0 +1,142 @@
+            Dict:
+                The resource record of the newly created target group.
+
+        Examples:
+            >>> tg = tio.target_groups.create('Example', ['192.168.0.0/24'])
+        """
+        schema = TargetGroupsSchema()
+        if not members:
+            members = []
+        members = ",".join(members)
+        payload = schema.load({"name": name, "members": members, **kw})
+        return self._post(json=payload)
+
+    def delete(self, id: int) -> None:
+        """
+        Delete a target group.
+
+        :devportal:`target-groups: delete <target-groups-delete>`
+
+        Args:
+            id (int): The unique identifier for the target group.
+
+        Returns:
+            None:
+                The target group was successfully deleted.
+
+        Examples:
+            >>> tio.target_groups.delete(1)
+        """
+        self._delete(id)
+
+    def details(self, id: int) -> Dict:
+        """
+        Retrieve the details of a target group.
+
+        :devportal:`target-groups: details <target-groups-details>`
+
+        Args:
+            id (int): The unique identifier for the target group.
+
+        Returns:
+            Dict:
+                The resource record for the target group.
+
+        Examples:
+            >>> tg = tio.target_groups.details(1)
+        """
+        return self._get(id)
+
+    def edit(self, id: int, **kw) -> Dict:
+        """
+        Edit an existing target group.
+
+        :devportal:`target-groups: edit <target-groups-edit>`
+
+        Args:
+            id (int): The unique identifier for the target group.
+            name (str, optional): The name of the target group.
+            members (list, optional):
+                The members of the target group.  FQDNs, CIDRs, IPRanges, and
+                individual IPs are supported.  NOTE: modifying the member list
+                is atomic and not additive.  All previous members that are
+                desired to be kept within the member list much also be included.
+            acls (list, optional):
+                A list of ACLs defining how the asset list can be used.  For
+                further information on how the ACL dictionaries should be
+                written, please refer to the API documentation.  NOTE: modifying
+                ACLs is atomic and not additive.  Please provide the complete
+                list of ACLs that this asset group will need.
+
+        Returns:
+            Dict:
+                The modified target group resource record.
+
+        Examples:
+            >>> tio.target_groups.edit(1, name='Updated TG Name')
+        """
+        # We need to get the current asset group and then merge in the modified
+        # data.  We will store the information in the same variable as the
+        # modified information was built into.
+        schema = TargetGroupsSchema(only=("name", "members", "acls"))
+        payload = schema.load(kw)
+        craw = schema.dump(self.details(id))
```

Comment:
And we will need to only pass 3 fields to schema.load, 
current = {
            'name': craw.get('name'),
            'acls': craw.get('acls'),
            'members': craw.get('members'),
        }
This will raise an exception in Marshmallow.

Reply:
schema.dump() will not raise any exception when passed with only all fields when in reality we need only 3 fields because of how the schema object is instansiated. Whereas schema.load() will result into exception when given all 4